dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantMemoryStore.cs

```diff
@@ -0,0 +1,369 @@
+            if (record != null)
+            {
+                yield return record;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Get a MemoryRecord from the Qdrant Vector database by pointId.
+    /// </summary>
+    /// <param name="collectionName"></param>
+    /// <param name="pointId"></param>
+    /// <param name="cancel"></param>
+    /// <returns></returns>
+    /// <exception cref="VectorDbException"></exception>
+    public async Task<MemoryRecord?> GetWithPointIdAsync(string collectionName, string pointId, CancellationToken cancel = default)
+    {
+        try
+        {
+            var vectorDataList = this._qdrantClient
+                .GetVectorsByIdAsync(collectionName, new[] { pointId }, cancel: cancel);
+
+            var vectorData = await vectorDataList.FirstOrDefaultAsync(cancel);
+
+            if (vectorData != null)
+            {
+                return MemoryRecord.FromJson(
+                    json: vectorData.GetSerializedPayload(),
+                    embedding: new Embedding<float>(vectorData.Embedding));
+            }
+            else
+            {
+                return null;
+            }
+        }
+        catch (Exception ex)
+        {
+            throw new VectorDbException($"Failed to get vector data from Qdrant {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Get a MemoryRecord from the Qdrant Vector database by given a group of pointIds.
+    /// </summary>
+    /// <param name="collectionName"></param>
+    /// <param name="pointIds"></param>
+    /// <param name="cancel"></param>
+    /// <returns></returns>
+    public async IAsyncEnumerable<MemoryRecord> GetWithPointIdBatchAsync(string collectionName, IEnumerable<string> pointIds,
+        [EnumeratorCancellation] CancellationToken cancel = default)
+    {
+        var vectorDataList = this._qdrantClient
+            .GetVectorsByIdAsync(collectionName, pointIds, cancel: cancel);
+
+        await foreach (var vectorData in vectorDataList)
+        {
+            yield return MemoryRecord.FromJson(
+                json: vectorData.GetSerializedPayload(),
+                embedding: new Embedding<float>(vectorData.Embedding),
+                key: vectorData.PointId);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task RemoveAsync(string collectionName, string key, CancellationToken cancel = default)
+    {
+        try
+        {
+            await this._qdrantClient.DeleteVectorByPayloadIdAsync(collectionName, key, cancel: cancel);
+        }
+        catch (Exception ex)
+        {
+            throw new VectorDbException($"Failed to remove vector data from Qdrant {ex.Message}");
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task RemoveBatchAsync(string collectionName, IEnumerable<string> keys, CancellationToken cancel = default)
+    {
+        foreach (var key in keys)
+        {
+            await this.RemoveAsync(collectionName, key, cancel);
```

Comment:
Couldn't this be made faster by not awaiting for each serially, but rather with a Task.WhenAll()?

Reply:
done, changed for the other vector stores where applicable as well.